includes/rest-api/endpoints/class-wp-rest-abilities-run-controller.php

```diff
@@ -163,7 +163,7 @@ public function run_ability_permissions_check( $request ) {
 		}
 
 		$input = $this->get_input_from_request( $request );
-		if ( ! $ability->has_permission( $input ) ) {
+		if ( ! $ability->check_permission( $input ) ) {
```

Comment:
Are we intentionally only handling false and saving the WP_Errors for `->execute()`? cc @emdashcodes 

```suggestion
		if ( true !== $ability->check_permission( $input ) ) {
```

Reply:
Let's fix it. I don't think it was intentional.

Reply:
In my testing, the proposed code changes fail the result of 4 tests in the run controller. We would have to better handle the reason the request errored so it doesn't default to 403. It probably means, we would have to move some logic to `run_ability_permissions_check`.

Reply:
An alternative would be to remove `run_ability_permissions_check()` (skip `permission_callback`) in the run controller, taking into account that `run_ability` (part of the `callback`) validates permissions anyway. It still would have to handle 403 when `check_permissions()` from the ability returns `false`.

Technically speaking, we are good. However, we can improve the implementation/documentation to avoid future misunderstandings.